addon/addon-test-support/@ember/test-helpers/-internal/build-registry.ts

```diff
@@ -57,13 +69,16 @@ const Owner = EmberObject.extend(RegistryProxyMixin, ContainerProxyMixin, {
    * @see {@link https://github.com/emberjs/ember.js/pull/12680}
    * @see {@link https://github.com/emberjs/ember.js/blob/v4.5.0-alpha.5/packages/%40ember/engine/instance.ts#L152-L167}
    */
-  unregister(fullName: string) {
-    // @ts-expect-error
-    this['__container__'].reset(fullName);
+  /* eslint-enable valid-jsdoc */
+  unregister(this: Owner, fullName: FullName) {
+    // SAFETY: this is always present, but only the stable type definitions from
+    // Ember actually preserve it, since it is private API.
```

Comment:
For reference: we cannot `ts-expect-error` this because it *actually works* on the stable types. 😂